From 5e8dbf9fa0d98acb1845b78800744e6e334ea69d Mon Sep 17 00:00:00 2001
From: "GONG, Ruiqi" <gongruiqi1@huawei.com>
Date: Fri, 6 Jan 2023 11:17:47 +0800
Subject: [PATCH 1265/1398] [Huawei] Revert "[Huawei] selinux: use GFP_ATOMIC
 in convert_context()"

Offering: HULK
hulk inclusion
category: bugfix
bugzilla: 188217

----------------------------------------

This reverts commit 2ebc5db0abb5 ("selinux: use GFP_ATOMIC in convert
context()"), which was a temporary bugfix.

The corresponding upstream bugfix patch, commit 2723875e9d67 ("selinux:
enable use of both GFP_KERNEL and GFP_ATOMIC in convert_context()") from
stable 5.10, will be back-ported.

Signed-off-by: GONG, Ruiqi <gongruiqi1@huawei.com>
---
 security/selinux/ss/services.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/security/selinux/ss/services.c b/security/selinux/ss/services.c
index a4977376aa09..31d631fa846e 100644
--- a/security/selinux/ss/services.c
+++ b/security/selinux/ss/services.c
@@ -2025,7 +2025,7 @@ static int convert_context(struct context *oldc, struct context *newc, void *p)
 	args = p;
 
 	if (oldc->str) {
-		s = kstrdup(oldc->str, GFP_ATOMIC);
+		s = kstrdup(oldc->str, GFP_KERNEL);
 		if (!s)
 			return -ENOMEM;
 
-- 
2.17.1

